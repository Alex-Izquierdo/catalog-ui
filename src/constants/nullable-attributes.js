export const PORTFOLIO_ITEM_NULLABLE = [
  'display_name', 'description', 'long_description', 'distributor', 'documentation_url', 'support_url', 'workflow_ref'
];
