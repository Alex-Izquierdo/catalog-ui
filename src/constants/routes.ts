export const ADD_PORTFOLIO_ROUTE = '/portfolios/add-portfolio';
export const ADD_PRODUCTS_ROUTE = '/portfolios/add-products';
export const EDIT_ORDER_PROCESS_ROUTE = '/portfolios/edit-order-process';
export const EDIT_PORTFOLIO_ROUTE = '/portfolios/edit';
export const EDIT_PORTFOLIO_WORKFLOW_ROUTE = '/portfolios/edit-workflow';
export const EDIT_SURVEY_ROUTE = '/portfolios/portfolio-item/edit-survey';
export const NESTED_EDIT_ORDER_PROCESS_ROUTE =
  '/portfolios/portfolio/edit-order-process';
export const NESTED_EDIT_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/edit-portfolio';
export const NESTED_REMOVE_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/remove-portfolio';
export const NESTED_SHARE_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/share-portfolio';
export const NESTED_WORKFLOW_PORTFOLIO_ROUTE =
  '/portfolios/portfolio/edit-workflow';
export const ORDER_ROUTE = '/orders/order';
export const ORDERS_ROUTE = '/orders';
export const PLATFORM_INVENTORIES_ROUTE = '/platforms/platform-inventories';
export const PLATFORM_ROUTE = '/platforms/platform';
export const PLATFORM_SERVICE_OFFERINGS_ROUTE = '/platforms/service-offerings';
export const PLATFORM_TEMPLATES_ROUTE = '/platforms/platform-templates';
export const PLATFORM_DETAILS_ROUTE = '/platforms/platform-details';
export const PLATFORMS_ROUTE = '/platforms';
export const PORTFOLIO_ROUTE = '/portfolios/portfolio';
export const PORTFOLIO_ITEM_ROUTE = '/portfolios/portfolio-item';
export const PORTFOLIO_ITEM_ORDER_ROUTE = '/portfolios/portfolio-item/order';
export const PORTFOLIO_ITEM_ROUTE_EDIT = '/portfolios/portfolio-item/edit';
export const PORTFOLIO_ITEM_EDIT_ORDER_PROCESS_ROUTE =
  '/portfolios/portfolio-item/edit-order-process';
export const PORTFOLIOS_ROUTE = '/portfolios';
export const REMOVE_PORTFOLIO_ROUTE = '/portfolios/remove';
export const SHARE_PORTFOLIO_ROUTE = '/portfolios/share';
export const WORKFLOW_PORTFOLIO_ROUTE = '/portfolios/edit-workflow';
export const ORDER_PROCESSES_ROUTE = '/order-processes';
export const ADD_ORDER_PROCESS_ROUTE = '/order-processes/add-order-process';
export const REMOVE_ORDER_PROCESS_ROUTE = '/order-processes/remove';
export const UPDATE_ORDER_PROCESS_ROUTE = '/order-processes/update';
