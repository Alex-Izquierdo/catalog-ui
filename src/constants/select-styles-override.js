const style = {
  container: provided => ({
    ...provided,
    '&:focus': {
      outline: 'none'
    }
  }),
  control: (provided, { isFocused }) => ({
    ...provided,
    borderRadius: 0,
    borderColor: '#EDEDED',
    backgroundColor: '#FFFFFF',
    borderBottomColor: isFocused ? '#007bba' : '#282d33',
    cursor: 'pointer',
    boxShadow: 'none',
    minHeight: 36,
    width: 256,
    '&:hover': {
      borderBottomColor: '#007bba'
    }
  }),
  indicatorSeparator: () => ({}),
  dropdownIndicator: provided => ({
    ...provided,
    color: '#282d33',
    paddingBottom: 6,
    paddingTop: 6
  }),
  menu: provided => ({
    ...provided,
    borderRadius: 'none',
    boxShadow:
      '0 0.0625rem 0.0625rem 0rem rgba(3,3,3,.05),0 0.25rem 0.5rem 0.25rem rgba(3,3,3,.06)',
    paddingBottom: '0.5rem',
    paddingTop: '0.5rem',
    marginTop: 4,
    width: 'calc(100% - 4px)',
    marginRight: 2,
    marginLeft: 2
  }),
  option: (provided, { isFocused }) => ({
    ...provided,
    cursor: 'pointer',
    textAlign: 'left',
    color: '#282d33',
    fontSize: '1rem',
    fontWeight: '400',
    lineHeight: '1.5',
    paddingBottom: '0.5rem',
    paddingTop: '0.5rem',
    paddingLeft: '1rem',
    paddingRight: '1rem',
    backgroundColor: isFocused ? '#ededed' : 'transparent',
    '&:hover': {
      backgroundColor: '#ededed'
    }
  }),
  placeholder: provided => ({
    ...provided,
    left: 20
  }),
  singleValue: provided => ({
    ...provided,
    left: 20,
    maxWidth: 'calc(100% - 28px)'
  })
};

export default style;
