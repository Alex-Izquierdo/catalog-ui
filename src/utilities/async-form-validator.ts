import awesomeDebouncePromise, {
  AwesomeDebounceOptions
} from 'awesome-debounce-promise';

const asyncFormValidator = (
  asyncFunction: (...args: any[]) => any,
  debounceTime = 250,
  options: Partial<AwesomeDebounceOptions> = { onlyResolvesLast: false }
): any => awesomeDebouncePromise(asyncFunction, debounceTime, options);

export default asyncFormValidator;
