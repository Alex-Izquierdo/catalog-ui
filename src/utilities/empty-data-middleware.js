const emptyDataMiddleware = () => (dispatch) => (action) => {
  const nextAction = { ...action };
  if (
    action.type.match(/_FULFILLED$/) &&
    action.payload &&
    action.meta &&
    action.payload.data &&
    action.payload.meta
  ) {
    nextAction.payload.meta.noData =
      nextAction.payload.meta.count === 0 &&
      nextAction.meta.filter.length === 0;
    return dispatch(nextAction);
  }

  return dispatch(nextAction);
};

export default emptyDataMiddleware;
