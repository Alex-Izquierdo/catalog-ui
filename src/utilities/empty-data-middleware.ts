import { Dispatch } from 'redux';
import { ReduxAction } from '../types/common-types';

const emptyDataMiddleware = () => (dispatch: Dispatch) => (
  action: ReduxAction
): ReduxAction => {
  const nextAction = { ...action };
  if (
    action.type.match(/_FULFILLED$/) &&
    action.payload &&
    action.meta &&
    action.payload.data &&
    action.payload.meta
  ) {
    const noFilter = Object.prototype.hasOwnProperty.call(
      nextAction.meta,
      'filters'
    )
      ? Object.values(nextAction.meta?.filters || {}).every(
          (value?: string) => typeof value === 'undefined' || value.length === 0
        )
      : nextAction.meta?.filter?.length === 0;

    nextAction.payload.meta.noData =
      nextAction.payload.meta.count === 0 && noFilter;
    return dispatch(nextAction);
  }

  return dispatch(nextAction);
};

export default emptyDataMiddleware;
