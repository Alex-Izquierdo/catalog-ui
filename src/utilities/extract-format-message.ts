import { I18nFormatMessage } from '../redux/reducers/i18n-reducer';
import { GetReduxState } from '../types/redux';

const extractFormatMessage = (getState: GetReduxState): I18nFormatMessage =>
  getState().i18nReducer.formatMessage!;

export default extractFormatMessage;
