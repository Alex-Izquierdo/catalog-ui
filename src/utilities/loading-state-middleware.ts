import { SET_LOADING_STATE } from '../redux/action-types';
import { Dispatch } from 'redux';
import { ReduxAction } from '../types/common-types';

const loadingStateMiddleware = () => (dispatch: Dispatch) => (
  action: ReduxAction
): ReduxAction => {
  if (action.type.match(/_REJECTED$/)) {
    dispatch({ type: SET_LOADING_STATE, payload: false });
  }

  return dispatch(action);
};

export default loadingStateMiddleware;
