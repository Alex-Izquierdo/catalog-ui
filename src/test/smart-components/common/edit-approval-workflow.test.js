import React from 'react';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store' ;
import { Modal } from '@patternfly/react-core';
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';
import promiseMiddleware from 'redux-promise-middleware';
import { componentTypes } from '@data-driven-forms/react-form-renderer';
import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications/';
import { APPROVAL_API_BASE } from '../../../utilities/constants';
import FormRenderer from '../../../smart-components/common/form-renderer';
import EditApprovalWorkflow from '../../../smart-components/common/edit-approval-workflow';

describe('<EditApprovalWorkflow />', () => {
  let initialProps;
  let initialState;
  const middlewares = [ thunk, promiseMiddleware(), notificationsMiddleware() ];
  let mockStore;
  const ComponentWrapper = ({ store, children, portfolioId }) => (
    <Provider store={ store }>
      <MemoryRouter initialEntries={ [ `portfolios/${portfolioId}` ] }>
        { children }
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {
      fetchWorkflows: jest.fn(),
      closeUrl: 'foo',
      objectType: 'Portfolio',
      workflows: [{ name: 'workflow', id: '123' }]
    };
    initialState = {
      approvalReducer: {
        workflows: [{
          label: 'foo',
          value: 'bar'
        }]
      }
    };
    mockStore = configureStore(middlewares);
  });

  it('should render correctly', () => {
    const store = mockStore({});
    apiClientMock.get(`${APPROVAL_API_BASE}/workflows`, mockOnce({ body: { data: []}}));
    const wrapper = shallow(<ComponentWrapper store={ store }><EditApprovalWorkflow { ...initialProps } /></ComponentWrapper>).dive();

    setImmediate(() => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });

  it('should create the edit workflow modal', done => {
    const store = mockStore(initialState);

    apiClientMock.get(`${APPROVAL_API_BASE}/workflows`, mockOnce({
      body: {
        data: [{
          name: 'workflow',
          id: '123'
        }]
      }
    }));
    apiClientMock.post(`${APPROVAL_API_BASE}/workflows/resolve`, mockOnce({ body: { name: 'workflow', id: '123' }}));

    const expectedSchema = {
      fields: [{
        component: componentTypes.SELECT,
        name: 'workflow',
        label: 'Approval workflow',
        loadOptions: expect.any(Function),
        isSearchable: true,
        isClearable: true
      }]
    };

    const wrapper = mount(
      <ComponentWrapper store={ store } portfolioId="123">
        <Route path="portfolios/:id?" render={ () => <EditApprovalWorkflow { ...initialProps } match={ { params: { id: '123' }} } /> }/>
      </ComponentWrapper>
    );

    setImmediate(() => {
      const modal = wrapper.find(Modal);
      const form = wrapper.find(FormRenderer);
      expect(modal.props().title).toEqual('Set approval workflow');
      expect(form.props().schema).toEqual(expectedSchema);
      done();
    });
  });
});
