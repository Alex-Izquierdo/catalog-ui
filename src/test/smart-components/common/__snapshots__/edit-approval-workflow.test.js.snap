// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditApprovalWorkflow /> should create the edit workflow modal 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <MemoryRouter>
    <EditApprovalWorkflow
      closeUrl="foo"
      objectType="Portfolio"
    />
  </MemoryRouter>
</ContextProvider>
`;
